src/agentscope/service/service_factory.py

```diff
@@ -50,17 +60,365 @@ def _get_type_str(cls: Any) -> Optional[Union[str, list]]:
+                )
+
+            # Type error for the arguments
+            if not isinstance(sub_cmd["arguments"], dict):
+                raise FunctionCallFormatError(
+                    "Except a dictionary for the arguments, but got "
+                    f"{type(sub_cmd['arguments'])} instead.",
+                )
+
+            # Leaving the type checking and required checking to the runtime
+            # error reporting during execution
+        return cmds
+
+    def _execute_func(self, cmds: List[dict]) -> str:
+        """Execute the function with the arguments.
+
+        Args:
+            cmds (`List[dict]`):
+                A list of dictionaries, where each dictionary contains the
+                name of the function and its arguments, e.g. {"name": "func1",
+                "arguments": {"arg1": 1, "arg2": 2}}.
+
+        Returns:
+            `str`: The prompt of the execution results.
+        """
+
+        execute_results = []
+        for i, cmd in enumerate(cmds):
+            func_name = cmd["name"]
+            service_func = self.service_funcs[cmd["name"]]
+            kwargs = cmd.get("arguments", {})
+
+            print(f">>> Executing function {func_name} with arguments:")
+            for key, value in kwargs.items():
+                value = (
+                    value if len(str(value)) < 50 else str(value)[:50] + "..."
+                )
+                print(f">>> \t{key}: {value}")
+
+            # Execute the function
+            try:
+                func_res = service_func.processed_func(**kwargs)
+            except Exception as e:
+                func_res = ServiceResponse(
+                    status=ServiceExecStatus.ERROR,
+                    content=str(e),
+                )
+
+            print(">>> END ")
+
+            status = (
+                "SUCCESS"
+                if func_res.status == ServiceExecStatus.SUCCESS
+                else "FAILED"
+            )
+
+            arguments = [f"{k}: {v}" for k, v in kwargs.items()]
+
+            execute_res = self.execute_result_template.format_map(
+                {
+                    "index": i + 1,
+                    "function_name": cmd["name"],
+                    "arguments": "\n\t\t".join(arguments),
+                    "status": status,
+                    "result": func_res.content,
+                },
+            )
+
+            execute_results.append(execute_res)
+
+        execute_results_prompt = "\n".join(execute_results)
+
+        return execute_results_prompt
+
+    def parse_and_call_func(self, text_cmd: Union[list[dict], str]) -> str:
+        """Parse, check the text and call the function."""
+
+        # --- Step 1: Parse the text according to the tools_call_format
+        cmds = self._parse_and_check_text(text_cmd)
+
+        # --- Step 2: Call the service function ---
+
+        execute_results_prompt = self._execute_func(cmds)
+
+        return execute_results_prompt
+
     @classmethod
     def get(
```

Comment:
It would be better if the example is added. (in Line 449)

```
Example:
```

Reply:
None